src/com/palmergames/bukkit/towny/event/town/TownKickEvent.java

```diff
@@ -0,0 +1,63 @@
+package com.palmergames.bukkit.towny.event.town;
+
+import com.palmergames.bukkit.towny.object.Resident;
+import com.palmergames.bukkit.towny.object.Translation;
+
+import org.bukkit.Bukkit;
+import org.bukkit.event.Cancellable;
+import org.bukkit.event.Event;
+import org.bukkit.event.HandlerList;
+
+/**
+ * Event that gets fired when a resident gets kicked from a town.
+ */
+public class TownKickEvent extends Event implements Cancellable {
+    private static final HandlerList handlers = new HandlerList();
+    private final Resident kickedResident;
+    private final Object kicker;
+    private boolean cancelled = false;
+	private String cancelMessage = Translation.of("msg_err_command_disable");
+
+    public TownKickEvent(Resident kickedResident, Object kicker) {
+        super(!Bukkit.getServer().isPrimaryThread());
+        this.kickedResident = kickedResident;
+        this.kicker = kicker;
+    }
+
+    public Resident getKickedResident() {
+        return kickedResident;
+    }
+
+    /**
+     * Gets whoever kicked the resident. Can either be an instance of Player or CommandSender.
+     * @return The kicker.
+     */
+    public Object getKicker() {
+        return kicker;
+    }
+
```

Comment:
Might be nice to have a .getTown() convenience method here, to avoid the user of the event having to do a the event.getKickedResident().getTown() which will require a NRE try/catch.